[V3] test downloader

bobloy opened the pull request:
### Type

- [x] Bugfix
- [ ] Enhancement
- [ ] New feature

### Description of the changes

Option for fixing #1663

bobloy committed:
ignore idea

bobloy committed:
Merge branch 'V3/bob-develop' of https://github.com/bobloy/Red-DiscordBot into V3/bob-develop

bobloy committed:
windows option

bobloy committed:
Merge branch 'V3/develop' into V3/test_downloader

bobloy committed:
Remove personal setting

bobloy committed:
Merge branch 'V3/test_downloader' of https://github.com/bobloy/Red-DiscordBot into V3/test_downloader

bobloy committed:
proper undo

tekulvw commented on tests/cogs/downloader/test_downloader.py:
```diff
@@ -96,7 +96,7 @@ def test_existing_git_repo(tmpdir):
     assert clone_cmd[3] == "rewrite_cogs"
     assert clone_cmd[4] == repo_norun.url
-    assert "repos/squid" in clone_cmd[5]
+    assert "repos/squid" in clone_cmd[5] or "repos\\\\squid" in clone_cmd[5]
```
This should compare against path parents.

For example:
```python
assert ["repos", "squid"] == pathlib.Path(clone_cmd[5]).parents[-2:]
```

bobloy committed:
Requested changes

```
("repos", "squid") == pathlib.Path('TEST\\\\TEST2\\\\repos\\\\squid').parts[-2:]
True
("repos", "squid") == pathlib.Path('test/test2/test3/test4/repos/squid').parts[-2:]
True
```

bobloy commented:
Commited changes to address review. Used pathlib.Path().parts

bobloy committed:
Needs to remove newline too

bobloy commented:
I have no idea what Travis is talking about with that build fail

Tobotimus committed:
Merge branch 'V3/develop' into V3/test_downloader

Tobotimus commented:
@bobloy We apologise for the confusion as Travis was installing a newer version of the stylechecking tool than the usual one. The issue has since been fixed, however we ask that you update your development requirements with `pipenv install --dev`.

Just as a side note, I would recommend running `pipenv install --dev` every time you pull/merge from V3/develop. Specifically, do it whenever you see that the `Pipfile.lock` file has been changed, for example here:

```bash
$ git pull
Updating 55afc7e..d3f406a
Fast-forward
 Pipfile.lock                           |  18 ++--
 ...
```

This usually means that a dependency has been pinned to a new version, and thus an update is necessary.

Tobotimus reviewed (approved):
Thanks for your contribution!
